scp/scpAgentProcessor.hpp

```diff
@@ -1,44 +1,40 @@
 /*
-* This source file is part of an OSTIS project. For the latest info, see http://ostis.net
-* Distributed under the MIT License
-* (See accompanying file COPYING.MIT or copy at http://opensource.org/licenses/MIT)
-*/
+ * This source file is part of an OSTIS project. For the latest info, see http://ostis.net
+ * Distributed under the MIT License
+ * (See accompanying file COPYING.MIT or copy at http://opensource.org/licenses/MIT)
+ */
 
 #pragma once
 
-#include "sc-memory/sc_addr.hpp"
-#include "sc-memory/sc_object.hpp"
-#include "sc-memory/kpm/sc_agent.hpp"
-#include "concurrent_queue.hpp"
-#include "scpAgentEvent.hpp"
-#include "scpKeynodes.hpp"
-
-#include "scpAgentProcessor.generated.hpp"
+#include <sc-memory/sc_agent.hpp>
 
 namespace scp
 {
 
-class ASCPAgentActivator : public ScAgent
+class ASCPAgentActivator : public ScAgent<ScEventAfterGenerateOutgoingArc<ScType::EdgeAccessConstPosPerm>>
 {
-    SC_CLASS(Agent, Event(Keynodes::active_sc_agent, ScEvent::Type::AddOutputEdge))
-    SC_GENERATED_BODY()
-
 public:
-    SC_PROPERTY(Keynode("sc_agent_of_scp_agents_activation"), ForceCreate)
-    static ScAddr msAgentKeynode;
+  ScAddr GetActionClass() const override;
+
+  ScResult DoProgram(ScEventAfterGenerateOutgoingArc<ScType::EdgeAccessConstPosPerm> const & event, ScAction & action)
+      override;
+
+  ScAddr GetEventSubscriptionElement() const override;
 
+  static inline ScKeynode const msAgentKeynode{"sc_agent_of_scp_agents_activation"};
```

Comment:
I think you can use this keynode to load specification from knowledge base within sc-module

Reply:
maybe later, I don't know what were they created for